File: TypeScript/creational/singleton-and-monostate/singleton/module-a.ts
import MyDatabase from './db/implementations/my-database-class'
// import MyDatabase from './db/implementations/my-database-object'
// import MyDatabase from './db/implementations/my-database-function'

const myDatabase = MyDatabase.instance
// const myDatabase = MyDatabase

myDatabase.store({
  name: 'Manoel',
  email: 'manoafonso.lopes@hotmail.com'
})
myDatabase.store({
  name: 'Athur Oliveira',
  email: 'aoliveira@hotmail.com'
})
myDatabase.store({
  name: 'Maju Coutinho',
  email: 'mjcoutinho@hotmail.com'
})

export default myDatabase

File: Python/creational/singleton-and-monostate/singleton/module_a.py
# from db.my_database import MyDatabase
from db.my_database_decorator import MyDatabaseDecorator as MyDatabase
# from db.my_database_metaclass import MyDatabaseMetaclass as MyDatabase

myDatabase = MyDatabase()

myDatabase.store({
    'name': 'Manoel',
    'email': 'manoafonso.lopes@hotmail.com'
})

myDatabase.store({
    'name': 'Athur Oliveira',
    'email': 'aoliveira@hotmail.com'
})

myDatabase.store({
    'name': 'Maju Coutinho',
    'email': 'mjcoutinho@hotmail.com'
})